test/test_decoders.py

```diff
@@ -1279,6 +1280,109 @@ def test_10bit_videos_cpu(self, asset):
+        ) as custom_frame_mappings:
+            decoder = VideoDecoder(
+                NASA_VIDEO.path,
+                stream_index=stream_index,
+                device=device,
+                custom_frame_mappings=custom_frame_mappings,
+            )
+            frame_0 = decoder.get_frame_at(0)
+            frame_5 = decoder.get_frame_at(5)
+            assert_frames_equal(
+                frame_0.data,
+                NASA_VIDEO.get_frame_data_by_index(0, stream_index=stream_index).to(
+                    device
+                ),
+            )
+            assert_frames_equal(
+                frame_5.data,
+                NASA_VIDEO.get_frame_data_by_index(5, stream_index=stream_index).to(
+                    device
+                ),
+            )
+            frames0_5 = decoder.get_frames_played_in_range(
+                frame_0.pts_seconds, frame_5.pts_seconds
+            )
+            assert_frames_equal(
+                frames0_5.data,
+                NASA_VIDEO.get_frame_data_by_range(0, 5, stream_index=stream_index).to(
+                    device
+                ),
+            )
+
+    @pytest.mark.parametrize("device", all_supported_devices())
+    def test_custom_frame_mappings_init_fails(self, tmp_path, device):
+        # Init fails if "approximate" seek mode is used with custom frame mappings
+        with pytest.raises(ValueError, match="seek_mode"):
+            VideoDecoder(
+                NASA_VIDEO.path,
+                stream_index=0,
+                device=device,
+                seek_mode="approximate",
+                custom_frame_mappings=NASA_VIDEO.generate_custom_frame_mappings(3),
+            )
+        # Write an invalid JSON file for testing
+        invalid_json_path = tmp_path / "invalid_json"
+        with open(invalid_json_path, "w+") as f:
+            f.write("""'{"invalid": "json"'""")
+        # Init fails if invalid JSON bytes are passed in as custom frame mappings
+        with pytest.raises(
+            ValueError,
+            match="Invalid custom frame mappings. It should be a valid JSON string or a JSON file object.",
+        ):
+            with open(invalid_json_path, "r") as f:
+                VideoDecoder(
+                    NASA_VIDEO.path,
+                    stream_index=0,
+                    device=device,
+                    custom_frame_mappings=f,
+                )
+        # Init fails if invalid JSON string is passed in as custom frame mappings
+        invalid_json_path = tmp_path / "invalid_json"
+        with pytest.raises(
+            ValueError,
+            match="Invalid custom frame mappings. It should be a valid JSON string or a JSON file object.",
+        ):
+            with open(invalid_json_path, "r") as f:
+                VideoDecoder(
+                    NASA_VIDEO.path,
+                    stream_index=0,
+                    device=device,
+                    custom_frame_mappings=f.read(),
+                )
```

Comment:
Similar to the comments I left in `read_custom_frame_mappings()`, I think we should also test for valid JSON that does not follow the schema that FFmpeg produces.